frontend/src/features/properties/selector/search/PropertySelectorSearchContainer.tsx

```diff
@@ -66,16 +77,25 @@ export const PropertySelectorSearchContainer: React.FunctionComponent<IPropertyS
         result = await findByPlanNumber(layerSearch.planNumber);
       } else if (layerSearch?.searchBy === 'legalDescription' && layerSearch.legalDescription) {
         result = await findByLegalDescription(layerSearch.legalDescription);
+      } else if (layerSearch?.searchBy === 'address' && layerSearch.address) {
+        // Ignore address searches
+        return;
       }
 
       const foundProperties = featuresToIdentifiedMapProperty(result) ?? [];
       // match the region and district for all found properties
       if (result?.features?.length !== undefined && result?.features?.length <= 15) {
-        await Promise.all(
-          foundProperties.map(p => matchRegionAndDistrict(p, regionService, districtService)),
+        var matchTask = foundProperties.map(p =>
+          matchRegionAndDistrict(p, findOneWhereContainsRegion, findOneWhereContainsDistrict),
         );
 
-        await Promise.all(foundProperties.map(p => matchAddress(p, getNearestToPoint)));
+        var getAddressTasks = foundProperties.map(p => getPropertyAddress(p, getNearestToPoint));
```

Comment:
nit: might be able to increase performance here by using a promise chain on each getPropertyAddress call instead of getting all addresses in a loop and then looping again through the responses.

Reply:
we could revisit it. At this point I rather just wrap up the bug and do some more research on the chaining later.